affinidi-did-resolver-cache-sdk/src/networking/request_queue.rs

```diff
@@ -136,3 +136,206 @@ impl RequestList {
+        assert_eq!(request_list.total_count, 0);
+    }
+
+    #[tokio::test]
+    async fn insert_works_returns_true() {
+        let config = config::ClientConfigBuilder::default().build();
+        let mut request_list = RequestList::new(&config);
+
+        let (tx, _) = oneshot::channel::<WSCommands>();
+
+        let unique_id: String = _unique_id();
+        let did_hash = _hash_did(&DID_KEY);
+
+        let insert_result = request_list.insert(did_hash.clone(), &unique_id, tx);
+
+        assert!(insert_result);
+    }
+
+    #[tokio::test]
+    async fn insert_works_returns_false_duplicates() {
+        let config = config::ClientConfigBuilder::default().build();
+        let mut request_list = RequestList::new(&config);
+
+        let (tx, _) = oneshot::channel::<WSCommands>();
+        let (tx2, _) = oneshot::channel::<WSCommands>();
+
+        let unique_id: String = _unique_id();
+        let did_hash = _hash_did(DID_KEY);
+
+        let insert_result = request_list.insert(did_hash.clone(), &unique_id, tx);
+        let insert_result2 = request_list.insert(did_hash.clone(), &unique_id, tx2);
+
+        assert!(insert_result);
+        assert_eq!(insert_result2, false);
+    }
+
+    #[tokio::test]
+    async fn insert_list_becomes_full() {
+        let config = config::ClientConfigBuilder::default()
+            .with_network_cache_limit_count(1)
+            .build();
+        let mut request_list = RequestList::new(&config);
+
+        let (tx, _) = oneshot::channel::<WSCommands>();
+        let (tx2, _) = oneshot::channel::<WSCommands>();
+
+        let unique_id: String = _unique_id();
+        let unique_id_2: String = _unique_id();
+
+        let did_hash = _hash_did(DID_KEY);
+        let did_hash_2 = _hash_did(DID_KEY_2);
+
+        let insert_result = request_list.insert(did_hash.clone(), &unique_id, tx);
+        let insert_result2 = request_list.insert(did_hash_2.clone(), &unique_id_2, tx2);
+
+        assert!(insert_result);
+        assert!(insert_result2);
+        assert!(request_list.list_full);
+
+        assert_eq!(request_list.total_count, 2);
+    }
+    #[should_panic]
+    #[tokio::test]
+    async fn remove_key_not_found() {
+        let config = config::ClientConfigBuilder::default().build();
+        let mut request_list = RequestList::new(&config);
+
+        request_list.remove(&_hash_did(DID_KEY), None).unwrap();
+    }
+
+    #[should_panic]
+    #[tokio::test]
+    async fn remove_key_not_found_passing_uuid() {
+        let config = config::ClientConfigBuilder::default().build();
+        let mut request_list = RequestList::new(&config);
+        request_list
+            .remove(&_hash_did(DID_KEY), Some("".to_string()))
+            .unwrap();
+    }
+
+    #[should_panic]
+    #[tokio::test]
+    async fn remove_key_not_found_passing_uuid_wrong_did() {
+        let config = config::ClientConfigBuilder::default().build();
+        let mut request_list = RequestList::new(&config);
+        request_list
```

Comment:
Maybe we can check for some specific the error, instead of having a check for panic?